modular_lumos/code/game/objects/items/fish_fish.dm

```diff
@@ -0,0 +1,245 @@
+
+/obj/item/fish_box/lobster
+	name = "lobster box"
+	inside_fish = /obj/item/fishy/lobster
+
+/obj/item/fish_box/catfish
+	name = "catfish box"
+	inside_fish = /obj/item/fishy/catfish
+
+/obj/item/fish_box/attack_self(mob/user)
+	if(inside_fish)
+		for(var/i in 1 to 2)
+			new inside_fish(get_turf(user))
+		qdel(src)
+
+/obj/item/fishy
+	name = "fish"
+	desc = "parent fish, do not use"
+	icon = 'modular_lumos/icons/obj/fish_items.dmi'
+
+	///Sex: either male or female
+	//Im sorry, but only male and female fish can breed together
+	var/sex = MALE_FISH
+
+	///Age: how old the fish is.
+	//Fish get older. They can't breed when young, and can't breed when old.
+	//as they get older, their max health goes down, until 0
+	//young age is a 1/4, middle age is 2/4s, and old age is a 1/4
+	var/age = 0
+
+	var/ageStatus = YOUNG_FISH
+
+	///In-tank: if the fish is in the tank, it doesn't get hurt.
+	var/in_tank = FALSE
+
+	///Health: maxhealth is the max health, health is the current health
+	//if a fish hits 0, it dies and cannot breed.
+	//if it is not hungry and inside a tank, it will gain health back
+	var/maxHealth = 100
+	var/health = 100
+
+	///Hunger: maxhunger is the max hunger, hunger is the current hunger
+	//if a fish hits 0, it starts losing health
+	//while starving, a fish will not produce. they will also age much faster
+	var/hunger = 100
+	var/maxHunger = 100
+
+	///Dead: the status if its dead.
+	//if dead, the tank will start to get dirty faster, and the other fish will start to lose health as well
+	var/dead = FALSE
+
+	//this is for the process to do every 5 seconds
+	var/timed_aging
+
+	///Spawned-Egg: what egg is spawned when there is a male and female that are ready to breed
+	var/obj/item/fishy_egg/spawned_egg
+
+	var/meat
+
+/obj/item/fishy/Initialize()
+	. = ..()
+	if(prob(50))
+		sex = MALE_FISH
+	else
+		sex = FEMALE_FISH
+	START_PROCESSING(SSobj, src)
+
+/obj/item/fishy/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/fishy/process()
+	if(dead)
+		return
+
+	if(world.time < timed_aging)
+		return
+
+	timed_aging = world.time + 5 SECONDS
+
+	age++
+	if(prob(age / 10))
+		switch(ageStatus)
+			if(YOUNG_FISH)
+				ageStatus = MIDDLE_FISH
+			if(MIDDLE_FISH)
+				ageStatus = OLD_FISH
+			if(OLD_FISH)
+				maxHealth--
+		age = 0
+
+	hunger--
+	if(hunger <= 0)
+		hunger = 0
+		health--
+	else if(hunger > 0)
+		health++
+
+	if(!in_tank)
+		health -= 2
+
+	if(health > maxHealth)
+		health = maxHealth
+	if(health <= 0)
+		health = 0
+		dead = TRUE
+		return
+	
+
+/obj/item/fishy/attackby(obj/item/I, mob/living/user, params)
+	if(I.get_sharpness())
+		to_chat(user, "<span class='notice'>You begin to butcher [I]...</span>")
+		playsound(I.loc, 'modular_skyrat/sound/effects/butcher.ogg', 50, TRUE, -1)
+		if(!do_after(user, 4 SECONDS, FALSE, src))
+			return
+		for(var/spawned_meat in meat)
+			for(var/i in 1 to rand(1,3))
+				new spawned_meat(get_turf(src))
+		new /obj/effect/gibspawner/generic(get_turf(src))
```

Comment:
```suggestion

```
I've played around with it, and when you start chopping up more than one fish, on a table, nonetheless, the whole kitchen ends up looking like a scene out of Cannibal Holocaust.

Reply:
fixed/removed